velodyne_gazebo_plugins/include/velodyne_gazebo_plugins/GazeboRosVelodyneLaser.h

```diff
@@ -61,12 +61,12 @@ namespace gazebo
       struct Sample {
         Sample() = default;
 
-        Sample(int _index, ignition::math::Angle _angle)
+        Sample(double _index, ignition::math::Angle _angle)
           : index(_index), angle(_angle)
         {
         }
 
-        int index{0};
+        double index{0.};
```

Comment:
> It's no longer really an "index" if it is a double, right? I think it makes sense to rename this variable to what it actually represents now.

I thought about it as a _"fractional"_ index, but I'm open to other names. It's a coordinate in grid axis units.

> We should change the for loop in SampleAngleIntervalEvenly to use doubles instead of integers. It's mostly pedantic (as the promotion happens automatically), but we may as well be as type-correct as we can be.

Fair enough. Thinking about it again, we won't have any precision issues when using doubles with integer values. See c73e6e2ac4b017822ea824ff94e082533aeed578.